tests/functional/openstackdataplanedeployment_controller_test.go

```diff
@@ -30,23 +38,116 @@ var _ = Describe("Dataplane Deployment Test", func() {
 			Namespace: namespace,
 			Name:      "dataplane-ansible-ssh-private-key-secret",
 		}
+		neutronOvnMetadataSecretName = types.NamespacedName{
+			Namespace: namespace,
+			Name:      "neutron-ovn-metadata-agent-neutron-config",
+		}
+		novaNeutronMetadataSecretName = types.NamespacedName{
+			Namespace: namespace,
+			Name:      "nova-metadata-neutron-config",
+		}
+		novaCellComputeConfigSecretName = types.NamespacedName{
+			Namespace: namespace,
+			Name:      "nova-cell1-compute-config",
+		}
 		err := os.Setenv("OPERATOR_SERVICES", "../../config/services")
 		Expect(err).NotTo(HaveOccurred())
 	})
 
 	When("A dataplaneDeployment is created with matching NodeSet", func() {
 		BeforeEach(func() {
-			DeferCleanup(th.DeleteInstance, CreateDataplaneNodeSet(dataplaneNodeSetName, DefaultDataPlaneNoNodeSetSpec()))
-			DeferCleanup(th.DeleteInstance, CreateDataplaneDeployment(dataplaneDeploymentName, DefaultDataPlaneDeploymentSpec()))
 			CreateSSHSecret(dataplaneSSHSecretName)
+			CreateSSHSecret(neutronOvnMetadataSecretName)
+			CreateSSHSecret(novaNeutronMetadataSecretName)
+			CreateSSHSecret(novaCellComputeConfigSecretName)
+			DeferCleanup(th.DeleteInstance, CreateDataplaneNodeSet(dataplaneNodeSetName, DefaultDataPlaneNodeSetSpec()))
+			DeferCleanup(th.DeleteInstance, CreateDataplaneDeployment(dataplaneDeploymentName, DefaultDataPlaneDeploymentSpec()))
 		})
 
 		It("should have conditions set", func() {
+			Eventually(func(g Gomega) {
+				nodeSet := dataplanev1.OpenStackDataPlaneNodeSet{}
+				baremetal := baremetalv1.OpenStackBaremetalSet{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      nodeSet.Name,
+						Namespace: nodeSet.Namespace,
+					},
+				}
+				// Create config map for OVN service
+				Eventually(func(g Gomega) {
+					ovnConfigMapName := types.NamespacedName{
+						Namespace: namespace,
+						Name:      "ovncontroller-config",
+					}
+					mapData := map[string]interface{}{
+						"ovsdb-config": "test-ovn-config",
+					}
+					th.CreateConfigMap(ovnConfigMapName, mapData)
+				}).Should(Succeed())
+
+				// Set node set deployment ready condition to False
+				Eventually(func(g Gomega) {
+					g.Expect(th.K8sClient.Get(th.Ctx, dataplaneNodeSetName, &nodeSet)).To(Succeed())
+					nodeSet.Status.Conditions.MarkFalse(
+						condition.Type(fmt.Sprintf(dataplanev1.NodeSetServiceDeploymentReadyCondition, nodeSet.Name, "download-cache")),
+						condition.RequestedReason,
+						condition.SeverityInfo,
+						fmt.Sprintf(dataplanev1.NodeSetServiceDeploymentReadyMessage, nodeSet.Name, "download-cache"))
+					g.Expect(th.K8sClient.Status().Update(th.Ctx, &nodeSet)).To(Succeed())
+				}).Should(Succeed())
+
+				// Set baremetal provisioning conditions to True
+				Eventually(func(g Gomega) {
+					g.Expect(th.K8sClient.Get(th.Ctx, dataplaneNodeSetName, &baremetal)).To(Succeed())
```

Comment:
to avoid confusion I would not reuse the dataplaneNodeSetName as the name of the baremetalset. In reality the name of the two CRs are the same, but reading the test code is confusing this way.